### ISSUE TITLE
Windows switching 

### ISSUE THREAD

**Issue by xvladqt** [paperwm/PaperWM#441]
Hi, i have a bug with paperwm after upgrade gnome to 43, when i use hotkeys for switch windows after release button they rollback to original window. 

Video demo - https://drive.google.com/file/d/1ORxH4GBS_-8ItRen61SBhUXlpJcllWCr/view

**Comment by mateotomas2** [comment]
Same problem here using super + left / right. Super + tab works.

**Comment by autumn-birds** [comment]
I am also having this issue!  Super+tab works for me as well, though it's not ideal.

It started to occur post-rebooting after a full system upgrade.  My setup is Arch on zfs, so I have a snapshot of how the system was previously. Here are the versions of gnome-shell on the main system and on the snapshot--

*main system:*
```
Name            : gnome-shell
Version         : 1:43.1-1
Description     : Next generation desktop shell
[...snip...]
Optional Deps   : gnome-control-center: System settings [installed]
                  evolution-data-server: Evolution calendar integration
                  [installed]
                  gst-plugins-good: Screen recording [installed]
                  gst-plugin-pipewire: Screen recording
                  gnome-bluetooth-3.0: Bluetooth support [installed]
                  power-profiles-daemon: Power profile switching
```

*pre-upgrade snapshot:*
```
Name            : gnome-shell
Version         : 1:43.0-2
Description     : Next generation desktop shell
[...snip...]
Optional Deps   : gnome-control-center: System settings [installed]
                  evolution-data-server: Evolution calendar integration
                  [installed]
                  gst-plugins-good: Screen recording [installed]
                  gst-plugin-pipewire: Screen recording
                  gnome-bluetooth-3.0: Bluetooth support [installed]
```

Hopefully this is helpful.  I'm going to look into whether downgrading just gnome-shell is possible and, if so, whether that helps.  My PaperWM appears to be up-to-date with current `develop`, except for some 'upstream' changes in README.md (I had manually merged in the supported-version bump PR for gnome shell 43.)

**Comment by autumn-birds** [comment]
update: downgrading gnome-shell alone didn't help.  Booting from a clone (like a fork) of the previous snapshot did, so I assume a change in something other than gnome-shell *might* be causing this.  I don't have time to debug further and will just try to get used to using super+tab for now, but if it would be helpful I might be able to look at potentially coming up with a list of what packages changed at some point.

**Comment by mateotomas2** [comment]
Could be this helps. I was using X11 but after testing it with Gnome Wayland this problem disappeared. But on Wayland I have random crashes while moving windows that makes it unusable.

**Comment by ajfpay** [comment]
even win+tab on manjaro latest, paperWM not work

**Comment by hikhvar** [comment]
Same problem here with Manjaro, Gnome 
```
gnome-shell --version
GNOME Shell 43.1
```

```
yay -Qi gnome-shell
Name                     : gnome-shell
Version                  : 1:43.1-2
```

**Comment by lessthanseventy** [comment]
Same problem here

```
 ╰─λ paru -Qi gnome-shell     
Name            : gnome-shell
Version         : 1:43.1-2
Description     : Next generation desktop shell
Architecture    : x86_64
URL             : https://wiki.gnome.org/Projects/GnomeShell
Licenses        : GPL
Groups          : gnome
Provides        : None
Depends On      : accountsservice  gcr-4  gjs  upower  gnome-session  gtk4  gnome-settings-daemon  gsettings-desktop-schemas  libcanberra-pulse  libgdm  libsecret  mutter  libnma  unzip  libibus  gnome-autoar
                  gnome-disk-utility  libsoup3  libgweather-4  libmutter-11.so=0-64
Optional Deps   : gnome-control-center: System settings [installed]
                  evolution-data-server: Evolution calendar integration [installed]
                  gst-plugins-good: Screen recording [installed]
                  gst-plugin-pipewire: Screen recording [installed]
                  gnome-bluetooth-3.0: Bluetooth support [installed]
                  power-profiles-daemon: Power profile switching [installed]
Required By     : gdm  gnome-browser-connector  gnome-shell-extension-appindicator  gnome-shell-extension-freon  gnome-shell-extension-gnome-ui-tune-git  gnome-shell-extension-paperwm-git  gnome-shell-extension-pop-shell-git
                  gnome-shell-extension-system-monitor-git  gnome-shell-extension-topicons-plus  gnome-shell-extensions
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 11.27 MiB
Packager        : Jan Alexander Steffens (heftig) <heftig@archlinux.org>
Build Date      : Sat 12 Nov 2022 08:56:14 AM MST
Install Date    : Mon 14 Nov 2022 09:17:08 AM MST
Install Reason  : Explicitly installed
Install Script  : No
Validated By    : Signature
```

**Comment by hikhvar** [comment]
Did a quick bisect. Using the gnome-42 branch with the changes to the metadata.json from here: https://github.com/paperwm/PaperWM/commit/db9d63302b593c7a663791b577a306d3f432e18a will result in the same problem. I'm using X11 as well.

**Comment by Thesola10** [comment]
@autumn-birds 
I'm on Arch Linux and GNOME 43.1-2 like you, ~~and cannot reproduce this issue. Are there logs in `journalctl` whenever you use the window switching keystroke?~~ okay, I'm on Wayland.

Try using the key shortcut with a terminal running `journalctl -f` to see if any messages show up

Also, try changing the keystroke in PaperWM -- maybe it's a conflict, a bit like #413

**Comment by hikhvar** [comment]
@Thesola10 are you on X or wayland. On wayland this still works, on X it's broken :/

**Comment by Thesola10** [comment]
ah, I'm on Wayland sorry

**Comment by ajfpay** [comment]
> @Thesola10 are you on X or wayland. On wayland this still works, on X it's broken :/

Try material shell, I just found out that there is a split feature similar to paperwm

**Comment by pfzetto** [comment]
Same Problem. If you press Esc before releasing Super it selects the correct window.

**Comment by dmytrobabich** [comment]
Same problem too.

**Comment by pfzetto** [comment]
Changing `opensNavigator` and `opensMinimap` to false in `registerMinimapAction` in `keybindings.js` seems to fix it, but I don't know if it breaks anything else.

**Comment by bobbyboy1710** [comment]
It's working, thank you.

**Comment by dmytrobabich** [comment]
Super+Backspace doesn't work correct

**Comment by mateotomas2** [comment]
@pfz4 It works also for me. But when I change from one window to another the focus is still on the first window.
Seems like the problem is about how PaperWM set the focus of the window.

[Screencast from 2022-11-29 15-56-13.webm](https://user-images.githubusercontent.com/36208432/204563106-308c6a74-9ccc-40fa-b7b6-cb8e883c4dc8.webm)

**Comment by mateotomas2** [comment]
After a lot of debugging I found a better solution. 
More info here: 
https://github.com/paperwm/PaperWM/pull/446

**Comment by projectoneuniverse** [comment]
> After a lot of debugging I found a better solution. More info here: #446

After implementing #446, sometimes if open ulauncher and cpupower-gui, gnome shell crash

![image](https://user-images.githubusercontent.com/50221511/207247946-0f35ac0d-0c13-427d-ad3b-bc785280db82.png)

I don't know exactly what happens, but in Gnome 43 have trouble like that

**Comment by jtaala** [comment]
> > After a lot of debugging I found a better solution. More info here: #446
> 
> After implementing #446, sometimes if open ulauncher and cpupower-gui, gnome shell crash
> 
> ![image](https://user-images.githubusercontent.com/50221511/207247946-0f35ac0d-0c13-427d-ad3b-bc785280db82.png)
> 
> I don't know exactly what happens, but in Gnome 43 have trouble like that

Thanks @projectoneuniverse, I believe the issue you mentioned is unrelated to the #446 but should be fixed by PR #449 - it's caused by a floating window trying to open another window (root cause looks like it's related to a recent change/regression in libmutter).  Maybe try merge in #449 and see if it resolves what you're seeing.

**Comment by jtaala** [comment]
Thanks all, and thanks to @mateotomas2 for providing the fix #446.  Closing this issue now.
